test/controllers/ContactDetailsControllerSpec.scala

```diff
@@ -16,31 +16,40 @@
 
 package controllers
 
-import controllers.actions.DataRetrievalAction
+import controllers.actions.{DataRetrievalAction, MrnRequiredAction}
 import forms.mappings.ContactDetailsMapping._
 import models._
 import models.requests.SignedInUser
 import org.mockito.ArgumentMatchers._
 import org.mockito.Mockito.{reset, when}
 import org.scalacheck.Arbitrary._
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
 import play.api.data.Form
 import play.api.test.Helpers._
+import play.api.test.Injecting
 import play.twirl.api.HtmlFormat
 import views.html.contact_details
 
-class ContactDetailsControllerSpec extends ControllerSpecBase {
+class ContactDetailsControllerSpec extends ControllerSpecBase with GuiceOneAppPerSuite with Injecting {
 
   private val form = Form(contactDetailsMapping)
   val page = mock[contact_details]
 
   val emailRegex = """^[a-zA-Z0-9\.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-.]+$"""
+  val mrn: MRN = arbitraryMrn.arbitrary.sample.get
 
   def view(form: Form[ContactDetails] = form): String = page(form)(fakeRequest, messages).toString
 
   def controller(signedInUser: SignedInUser, eori: String, dataRetrieval: DataRetrievalAction = new FakeDataRetrievalAction(None)) =
-    new ContactDetailsController(new FakeAuthAction(signedInUser), new FakeEORIAction(eori), dataRetrieval, mockAnswersConnector, mcc, page)(
-      mcc.executionContext
-    )
+    new ContactDetailsController(
+      new FakeAuthAction(signedInUser),
+      new FakeEORIAction(eori),
+      dataRetrieval,
+      inject[MrnRequiredAction],
```

Comment:
Why not create a 'FakeMrnRequiredAction' like the other two actions? Would save you having to extend GuiceOneAppPerSuite and Injecting in the test class. Also keep consistency of approach :slightly_smiling_face:

Reply:
Agree with @glidester . What is even more important, is that in order for injector to work, you need to start the application andthis takes significantly more time. Also, it is arguable if such test is a Unit Test anymore.

Reply:
Thanks, I've updated this now